src/wasm.h

```diff
@@ -924,6 +924,9 @@ class Module {
   std::vector<UserSection> userSections;
   std::vector<std::string> debugInfoFileNames;
 
+  FeatureSet features = FeatureSet::MVP;
+  bool emitFeaturesSection = false;
```

Comment:
Perhaps we should handle the features section the same way we handle the names section - we read it if it's there, but we don't emit it unless the user specifically asks us to?

Reply:
Hmm, it would make sense to unify what we do with the features section and the names section (and the producers section? I forget what we do there). OTOH, it is very useful to preserve the features section through long chains of wasm-opt calls such as when running the reducer. Otherwise all but the first call of wasm-opt would have a validation failure.

Reply:
I believe we decided offline to leave this how it is for now but to leave this open for unification with how we treat other sections in the future.

Reply:
(and rename to `hasFeaturesSection`)

Reply:
Right, thank you.